pkg/validations/createcluster/createcluster.go

```diff
@@ -0,0 +1,90 @@
+package createcluster
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/gitops/flux"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/providers/validator"
+	"github.com/aws/eks-anywhere/pkg/validations"
+)
+
+type validationManager struct {
+	clusterSpec       *cluster.Spec
+	provider          providers.Provider
+	gitOpsFlux        *flux.Flux
+	createValidations Validator
+	dockerExec        validations.DockerExecutable
+}
+
+type Validator interface {
+	PreflightValidations(ctx context.Context) error
+}
+
+func NewValidations(clusterSpec *cluster.Spec, provider providers.Provider, gitOpsFlux *flux.Flux, createValidations Validator, dockerExec validations.DockerExecutable) *validationManager {
+	return &validationManager{
+		clusterSpec:       clusterSpec,
+		provider:          provider,
+		gitOpsFlux:        gitOpsFlux,
+		createValidations: createValidations,
+		dockerExec:        dockerExec,
+	}
+}
+
+func (v *validationManager) Validate(ctx context.Context) error {
+	runner := validations.NewRunner()
+	runner.Register(v.generateCreateValidations(ctx)...)
+	runner.Register(v.gitOpsFlux.Validations(ctx, v.clusterSpec)...)
+	err := runner.Run()
+
+	return err
+}
+
+func (v *validationManager) generateCreateValidations(ctx context.Context) []validations.Validation {
+	return []validations.Validation{
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate docker executable",
+				Err:    validations.ValidateDockerExecutable(ctx, v.dockerExec, runtime.GOOS),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate kubeconfig path",
+				Err:    kubeconfig.ValidateKubeconfigPath(v.clusterSpec.Cluster.Name),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate cluster",
+				Err:    cluster.ValidateConfig(v.clusterSpec.Config),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name:   "validate supported provider",
+				Err:    validator.ValidateSupportedProviderCreate(v.provider),
+				Silent: true,
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name: "create preflight validations pass",
+				Err:  v.createValidations.PreflightValidations(ctx),
+			}
+		},
+		func() *validations.ValidationResult {
+			return &validations.ValidationResult{
+				Name: fmt.Sprintf("validate %s Provider", v.provider.Name()),
+				Err:  v.provider.SetupAndValidateCreateCluster(ctx, v.clusterSpec),
```

Comment:
do these two need to be silent as well?

Reply:
No. We didn't add Silent here to keep the output consistent with what is shown when running the create cmd